data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -94,6 +95,12 @@ protected DeleteFilter(String filePath, List<DeleteFile> deletes, Schema tableSc
     this.eqDeletes = eqDeleteBuilder.build();
     this.requiredSchema = fileProjection(tableSchema, requestedSchema, posDeletes, eqDeletes);
     this.posAccessor = requiredSchema.accessorForField(MetadataColumns.ROW_POSITION.fieldId());
+    this.hasColumnIsDeleted = requestedSchema.findField(MetadataColumns.IS_DELETED.fieldId()) != null;
```

Comment:
nit: I saw a lot of places in Iceberg just use `null` to indicate the condition. should we use null for `columnIsDeletedIndex` too?

Reply:
A boolean variable should be expressive.